java/src/jmri/jmrix/loconet/locomon/LocoMonPane.java

```diff
@@ -85,14 +85,14 @@ public synchronized void initComponents(LocoNetSystemConnectionMemo memo) {
     public synchronized void message(LocoNetMessage l) {  // receive a LocoNet message and log it
         // send the raw data, to display if requested
         String raw = l.toString();
-        //format the message text, expect it to provide consistent \n after each line
+        // format the message text, expect it to provide consistent \n after each line
         String formatted = llnmon.displayMessage(l);
 
         // display the formatted data in the monitor pane
         nextLine(formatted, raw);
 
-        //include loconet monitoring in session.log if TRACE enabled
-        log.trace(formatted.substring(0, formatted.length() - 1));
+        // include loconet monitoring in session.log if TRACE enabled
+        log.trace(formatted.substring(0, formatted.length() - 1));  // remove trailing newline
```

Comment:
All the prep for the log message (getting the substring if formatted) is done before evaluating if it should be logged. You might want to wrap this in a log.isTraceEnabled() test to avoid that overhead when not tracing.